### PR TITLE
Ignore reservedBucket checks for net/rpc requests

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Ignore reservedBucket checks for net/rpc requests

## Motivation and Context
All `net/rpc` requests go to `/minio`, so the existing
generic handler for reserved bucket check would essentially
erroneously send errors leading to distributed setups to
wait infinitely.

For `net/rpc` requests alone we should skip this check and
allow resource bucket names to be from `/minio` .

## How Has This Been Tested?
Manually using the following script.

```sh
#!/bin/bash

$(minio server --address localhost:9001 https://localhost:9001/tmp/1/1 https://localhost:9002/tmp/2/2 https://localhost:9003/tmp/3/3 https://localhost:9004/tmp/4/4)&
$(minio server --address localhost:9002 https://localhost:9001/tmp/1/1 https://localhost:9002/tmp/2/2 https://localhost:9003/tmp/3/3 https://localhost:9004/tmp/4/4)&
$(minio server --address localhost:9003 https://localhost:9001/tmp/1/1 https://localhost:9002/tmp/2/2 https://localhost:9003/tmp/3/3 https://localhost:9004/tmp/4/4)&
minio server --address localhost:9004 https://localhost:9001/tmp/1/1 https://localhost:9002/tmp/2/2 https://localhost:9003/tmp/3/3 https://localhost:9004/tmp/4/4
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/generic-handlers.go [hunk 1]
```diff
@@ -202,20 +210,22 @@ func (h cacheControlHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 
-func (h minioPrivateBucketHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	// For all non browser requests, reject access to 'minioReservedBucketPath'.
-	bucketName, _ := urlPath2BucketObjectName(r.URL)
-	if !guessIsBrowserReq(r) && (isMinioReservedBucket(bucketName) || isMinioMetaBucket(bucketName)) {
-		writeErrorResponse(w, ErrAllAccessDisabled, r.URL)
-		return
+func (h minioReservedBucketHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if !guessIsRPCReq(r) && !guessIsBrowserReq(r) {
+		// For all non browser requests, reject access to 'minioReservedBucketPath'.
```
Comment (fwessels) [inline cmd/generic-handlers.go]: Add non RPC requests too?

**Review by fwessels — COMMENTED** [review]
LGTM

File: cmd/generic-handlers.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/generic-handlers.go]: There are only two types of requests which require /minio one is rpc and another is browser so we allow /minio for those operations. For all other API operations we reject it.

**Review by fwessels — APPROVED** [review]
Tested and approved.

**Review by balamurugana — APPROVED** [review]
verified by running mint

**Pull request merged**